sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/Messages.java

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.eventhubs;
+
+import com.azure.core.util.logging.ClientLogger;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+/**
+ * I18n messages loaded from the messages.properties file located within the same package.
+ */
+public enum Messages {
+    ;
+    private static final ClientLogger LOGGER = new ClientLogger(Messages.class);
+    private static Properties properties;
+    private static final String MESSAGES_PROPERTIES_PATH = "com/azure/messaging/eventhubs/messages.properties";
+    public static final String PROCESS_SPAN_SCOPE_TYPE_ERROR = getMessage("PROCESS_SPAN_SCOPE_TYPE_ERROR");
+    public static final String MESSAGE_NOT_OF_TYPE = getMessage("MESSAGE_NOT_OF_TYPE");
+    public static final String REQUEST_VALUE_NOT_VALID = getMessage("REQUEST_VALUE_NOT_VALID");
+    public static final String EVENT_DATA_DOES_NOT_FIT = getMessage("EVENT_DATA_DOES_NOT_FIT");
+    public static final String CANNOT_SEND_EVENT_BATCH_EMPTY = getMessage("CANNOT_SEND_EVENT_BATCH_EMPTY");
+    public static final String ERROR_SENDING_BATCH = getMessage("ERROR_SENDING_BATCH");
+    public static final String FAILED_TO_CLAIM_OWNERSHIP = getMessage("FAILED_TO_CLAIM_OWNERSHIP");
+    public static final String LOAD_BALANCING_FAILED = getMessage("LOAD_BALANCING_FAILED");
+    public static final String EVENT_PROCESSOR_RUN_END = getMessage("EVENT_PROCESSOR_RUN_END");
+    public static final String FAILED_PROCESSING_ERROR_RECEIVE = getMessage("FAILED_PROCESSING_ERROR_RECEIVE");
+    public static final String FAILED_WHILE_PROCESSING_ERROR = getMessage("FAILED_WHILE_PROCESSING_ERROR");
+    public static final String FAILED_CLOSE_CONSUMER_PARTITION = getMessage("FAILED_CLOSE_CONSUMER_PARTITION");
+    public static final String ERROR_OCCURRED_IN_SUBSCRIBER_ERROR = getMessage("ERROR_OCCURRED_IN_SUBSCRIBER_ERROR");
+    public static final String EXCEPTION_OCCURRED_WHILE_EMITTING = getMessage("EXCEPTION_OCCURRED_WHILE_EMITTING");
+
+    private static synchronized Properties getProperties() {
+        if (properties != null) {
+            return properties;
+        }
+        properties = new Properties();
+        try (InputStream inputStream =
+                 Thread.currentThread().getContextClassLoader().getResourceAsStream(MESSAGES_PROPERTIES_PATH)) {
+            properties.load(inputStream);
+        } catch (IOException exception) {
+            LOGGER.error("Error loading message properties {}", Messages.class, exception); //NON-NLS
```

Comment:
What happens if we are unable to load the inputStream.. Do the messages still show? Should we be rethrowing this IOException? @JonathanGiles ?

Reply:
If the messages couldn't be loaded ...

- there will be a error log indicating the cause of the failure
- the message text will not be shown but the _key_ of the message will be and the application continues to run...

I would not recommend re-throwing because since this called within a static initializer it will cause NoClassDefFoundError or ExceptionInInitializerError.


@conniey could you please review again